connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/streaming/StreamingQuerySuite.scala

```diff
@@ -78,4 +78,44 @@ class StreamingQuerySuite extends RemoteSparkSession with SQLHelper {
       }
     }
   }
+
+  test("Streaming table API") {
+    withSQLConf(
+      "spark.sql.shuffle.partitions" -> "1" // Avoid too many reducers.
+    ) {
+      spark.sql("DROP TABLE IF EXISTS my_table")
+
+      withTempPath {
+        ckpt =>
+          val q1 = spark.readStream
+            .format("rate")
+            .load()
+            .writeStream
+            .option("checkpointLocation", ckpt.getCanonicalPath)
+            .toTable("my_table")
+
+          val q2 = spark.readStream
+            .table("my_table")
+            .writeStream
+            .format("memory")
+            .queryName("my_sink")
+            .start()
+
+          try {
+            q1.processAllAvailable()
+            q2.processAllAvailable()
+            eventually(timeout(10.seconds)) {
+              assert(q2.recentProgress.nonEmpty) // Query made progress.
```

Comment:
This does not mean if would have processed any data, right? We could move `assert(spark.table("my_sink").count() > 0)` also into this.